src/appsettings/settings.py

```diff
@@ -706,20 +772,29 @@ def __init__(self, name='', default='', required=False,
                 the setting's prefix (overrides ``AppSettings.Meta`` prefix).
             call_default (bool): whether to call the default (if callable).
             transform_default (bool): whether to transform the default value.
-            **checker_kwargs: keyword arguments passed to the checker.
+            validators (list of callables): list of additional validators to use.
+            min_length (int): minimum length of the iterable (included).
+            max_length (int): maximum length of the iterable (included).
+            empty (bool): whether empty iterable is allowed. Deprecated in favor of min_length.
         """
         super(StringSetting, self).__init__(
             name=name, default=default, required=required, prefix=prefix,
-            call_default=call_default, transform_default=transform_default,
-            checker=StringTypeChecker(**checker_kwargs))
+            call_default=call_default, transform_default=transform_default, validators=validators)
+        if empty is not None:
+            warnings.warn("Empty argument is deprecated, use min_length instead.", DeprecationWarning)
+            min_length = 1
+        if min_length is not None:
+            self.validators.append(MinLengthValidator(min_length))
+        if max_length is not None:
+            self.validators.append(MaxLengthValidator(max_length))
 
 
-class ListSetting(Setting):
+class ListSetting(IterableSetting):
     """List setting."""
 
-    def __init__(self, name='', default=list, required=False,
-                 prefix='', call_default=True, transform_default=False,
-                 **checker_kwargs):
+    default_validators = (TypeValidator(list), )
+
+    def __init__(self, name='', default=list, **kwargs):
```

Comment:
I noticed, most of them now override the `__init__` just to provide a different default value. I was hoping to solve that in subsequent PR. I've created #37, so I don't forget.

I have tried to use the `**kwargs` whenever possible as the number of arguments is quite large, and in this case I consider the duplication worse than missing introspection. In places where it's not used, new argument are appended.

Also I forgot about `*args`, so I wiil add them as well, in case somebody uses them.